src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -31,22 +42,29 @@ public class LicenseManager : IDisposable
         private readonly LicenseStatus _licenseStatus = new LicenseStatus();
         private readonly BuildNumber _buildInfo;
         private Timer _leaseLicenseTimer;
-        private long _lastSavedLicenseIndex;
         private RSAParameters? _rsaParameters;
         private readonly ServerStore _serverStore;
-        private readonly SemaphoreSlim _semaphoreSlim = new SemaphoreSlim(1);
+        private readonly SemaphoreSlim _leaseLicenseSemaphore = new SemaphoreSlim(1);
+        private readonly SemaphoreSlim _recalculateLicenseLimitsSemsphore = new SemaphoreSlim(1);
 
         public LicenseManager(ServerStore serverStore)
         {
             _serverStore = serverStore;
-
+            
             _buildInfo = new BuildNumber
             {
                 BuildVersion = ServerVersion.Build,
                 ProductVersion = ServerVersion.Version,
                 CommitHash = ServerVersion.CommitHash,
                 FullVersion = ServerVersion.FullVersion
             };
+
+            _serverStore.LicenseChanged += async(_, e) =>
```

Comment:
This is almost always a bad idea. This is an `async void`, no one is waiting on this.
Meaning that an error here will be thrown away